### PR TITLE
Update tool of pushing podspecs.

### PR DESCRIPTION
**Pull request by granluo**

### CODE REVIEW

**Comment by granluo** [comment]
Most changes are corresponding to the comments under
[#6333](https://github.com/firebase/firebase-ios-sdk/pull/6333)

File: scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift [hunk 1]
```diff
@@ -81,150 +89,195 @@ struct Shell {
   }
 }
 
+// Error types that could occurred when this tool is running.
```
Comment (paulb777) [inline scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift:92]: nit: mixed tenses - `can occur when this tool runs` or just `Error types` only

File: scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift [hunk 2]
```diff
@@ -81,150 +89,195 @@ struct Shell {
...
 enum SpecRepoBuilderError: Error {
+  // Error occurs when circular dependenies are detected and deps will be
```
Comment (paulb777) [inline scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift:94]: dependencies

File: scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift [hunk 3]
```diff
@@ -81,150 +89,195 @@ struct Shell {
...
+  // displayed.
   case circularDependencies(pods: Set<String>)
+  // Error occurs when there exist a spec failed to push to a spec repo. All
+  // specs failed to push should be displayed.
```
Comment (paulb777) [inline scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift:98]: specs that failed to push

File: scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift [hunk 4]
```diff
@@ -81,150 +89,195 @@ struct Shell {
...
     }
 
     for pod in pods {
-      if !podSpecDict.contains(pod) {
-        continue
-      }
-      let deps = getTargetedDeps(of: pod, from: podSpecDict)
+      guard specFiles.contains(pod) else { continue }
+      let deps = getTargetedDeps(of: pod, from: specFiles)
+      // parentDeps will have all dependencies the current pod support. If the
```
Comment (paulb777) [inline scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift:142]: supports

File: scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift [hunk 5]
```diff
@@ -81,150 +89,195 @@ struct Shell {
...
+      // current pod were in the parent dependnecies, that means it was tracked
```
Comment (paulb777) [inline scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift:143]: dependencies

**Pull request merged**